astropy/units/function/logarithmic.py

```diff
@@ -277,6 +277,62 @@ def __isub__(self, other):
         self._set_unit(new_unit)
         return self
 
+    def __mul__(self, other):
+        # Multiply by a float or a dimensionless quantity
```

Comment:
To avoid the strange `a*a` leading to a unit of `mag` instead of `mag2`, I guess try something like
```
# Normal function units can only handle multiplication if the
# physical unit is dimensionless, but logarithmic ones can also
# handle anything to which a unit can be raised. So, try that first.
# But not for Quantity (to avoid, e.g., `Jy ** float(Magnitude(2.5)) = Jy**(1/10)`)
if not isinstance(other, FunctionQuantity):
    try:
        new_unit = self.unit.physical_unit ** other
    except Exception:
        pass
    else:
        result = self.view(np.ndarray) * other
        return self._new_view(result, new_unit)

# Do standard check on whether we are dimensionless.
return super().__mul__(other)
```

Reply:
Actually, I'm usually a big fan of `try/except`, but perhaps here it is really cleaner to just do
```
if isinstance(other, numbers.Number):
    new_unit = self.unit.physical_unit ** other
    result = self.view(np.ndarray) * other
    return self._new_view(result, new_unit)
else:
    return super().__mul__(other)
```
What do you think?

Reply:
Actually, I like the `if instance()` better, because "explicit is better than implicit". The try/except/else also separates lines which belong together; that also was the reason to keep everything in the try block.

Also delegation to the parent class is probably a good idea (DRY).